packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts

```diff
@@ -59,19 +66,17 @@ export class BeaconWallet implements WalletProvider {
   }
 
   async getPKH() {
-    const account = await this.client.getActiveAccount();
-    if (!account) {
+    if (!this.account) {
```

Comment:
One concern:
I the dApp code calls getPkh immediately after connecting, but before the first `ACTIVE_ACCOUNT_SET` event is received, they will get an error. Not sure if this is actually possible, or even if it's possible if there is a better solution.

Reply:
I don't think that will happen since the even will be subscribed to in the constructor. So getPkh will likely never be called unless the client has been constructed, and in consequence, the `account` member variable will always be assigned a value. And if it doesn't it will throw an error, so at least it signals the user (or developer) that the wallet has not been connected, or the wallet info is incomplete. I hope that makes sense!

Reply:
While the event is subscribed in the constructor, it might not be triggered until a split-second later

Reply:
hmm, I figured it would be a rare occurrence. But do you have any suggestions to improve it then?